Perseus34x/optimize comparisons

perseus34x opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.

Typos / punctuation / trivial PRs are generally not accepted.

Contributors guide: https://github.com/succinctlabs/sp1/blob/dev/CONTRIBUTING.md

The contributors guide includes instructions for running rustfmt and building the
documentation.
-->

<!-- ** Please select "Allow edits from maintainers" in the PR Options ** -->

## Motivation

I have now made the logic for comparing numbers more efficiently. Previously, comparisons were more expensive because they always ran two "less than" checks (both a < b and b < a`).

## Solution

-  The updated logic only runs the specific checks needed for the operation. 
- "Less than" (`lt`) now only runs one check. "Greater than" (`gt`) also uses just one check by swapping the numbers and reusing the "less than" logic. 
- Only the operations that rely on equality (like `le, eq, ne`) still use both checks, as this is how they figure out if `a == b. 
- This change removes unnecessary steps, reduces the system's execution log, and hence is more efficient for proof generation.

## PR Checklist

- [ X ] Added Tests
- [ ] Added Documentation
- [ X ] Breaking changes

Commit:
reduce number of event for I32Gts, I32Lts

Commit:
reduce number of event for I32Gts, I32Lts,I32GtU, I32LtU

Commit:
reduce number of event for I32Gts, I32Lts,I32GtU, I32LtU

nagatoism reviewed (commented):
lgtm.

check the linter error and you are good to go

Commit:
Adopted Efficient le/ge Logic: The constraints for "less-than-or-equal" (le) and "greater-than-or-equal" (ge) have been simplified. Instead of being defined as a complex OR with equality (e.g., a >= b as (a > b) OR (a == b)), they are now constrained using a more efficient boolean NOT operation:
-a >= b is constrained as NOT (a < b)
-a <= b is constrained as NOT (b < a)
This aligns the AIR with executor-level optimizations and removes a set of redundant constraints that were previously defining the same logic in a different way.

Commit:
correct lint

nagatoism reviewed (approved):
much cleaner. great improvment

djadjka commented on crates/rwasm/executor/src/executor.rs:
```diff
@@ -892,47 +892,60 @@ impl<'a> Executor<'a> {
+                        self.record.lt_events.push(AluEvent { a: 1 - lt_res, ..gt_comp_event });
+                    }
+                    // Equality checks need to know if `b < c` and `c < b` are both false.
+                    Opcode::I32Eq | Opcode::I32Eqz | Opcode::I32Ne => {
```
I can be wrong, but looks like I32Eq and I32Eqz can be processed without **lt_events** , just using asssert_eq

nagatoism replied:
you need comparision to establish that a is not eq to be
